python.d/logind.chart.py

```diff
@@ -57,23 +57,23 @@ def _get_data(self):
         seats = list()
         data = self._get_raw_data()
 
-        for item in data:
-            fields = item.split()
-            if len(fields) == 3:
-                users['remote'].append(fields[2])
-                ret['sessions_remote'] += 1
-            elif len(fields) == 4:
-                users['graphical'].append(fields[2])
-                ret['sessions_graphical'] += 1
-                seats.append(fields[3])
-            elif len(fields) == 5:
-                users['console'].append(fields[2])
-                ret['sessions_console'] += 1
-                seats.append(fields[3])
-
-        ret['users_graphical'] = len(set(users['graphical']))
-        ret['users_console'] = len(set(users['console']))
-        ret['users_remote'] = len(set(users['remote']))
-        ret['seats'] = len(set(seats))
+        if data:
```

Comment:
Isn't python on of those languages where you "ask for forgiveness" instead of "asking for permission"? What I mean is that we should probably use `try ... except` and catch exception.

Reply:
This approach is immediately clear as to what the intent is, if there's data, process it, otherwise just return empty stats (kind of, I need to fix that a bit, but that just involves modifying the initialization of the returned dictionary).

In contrast, handling this properly with a try/except arrangement will be a bit more complicated, because we _need_ to return actual errors for any other exception (including TypeError's) that may occur here so that things stay easily debuggable.  In essence, the except clause would end up having to inspect the message associated with the caught exception to determine whether to actually ignore it or propagate it.

If it weren't for that, I'd absolutely agree.

Reply:
> Isn't python on of those languages where you "ask for forgiveness" instead of "asking for permission"?

It is on of the approaches. Depends on how likely/unlikely is exception and how is expensive the check (`except` is expensive).